src/cls/IPM/Utils/Update/Controller.cls

```diff
@@ -0,0 +1,85 @@
+Include %IPM.Common
+
+Class %IPM.Utils.Update.Controller
+{
+
+/// Iterates through a module and its dependencies and applies their update steps that haven't been run up through the current version
+/// Starts with the most dependent module and works its way backwards to the base module we are calling "update" on
+ClassMethod ApplyUpdatesInOrder(
+	module As %IPM.Storage.Module,
+	logFile As %Library.File = "",
+	verbose As %Boolean = 0)
+{
+    if verbose {
+        write !, "Acquiring lock on ^%IPM.Update("_module.Name_")"
+    }
+    $$$LockUpdateModule(module.Name)
+    if ('$test) {
+        quit
+    }
+
+    set tSC = $$$OK
+    set tInitTLevel = $tlevel
+    try {
+        tstart
+        // For each base module dependency
+        set processList = ..GetModulesToUpdateOrderedList(module)
+
+        set ptr = 0
+        while $listnext(processList, ptr, moduleName) {
+            set nextModule = ##class(%IPM.Storage.Module).NameOpen(moduleName)
```

Comment:
Should open with concurrency level of 0 here to not acquire exclusive lock since you are just reading from the object